### PR TITLE
Remaining precompiles v2

### PR DESCRIPTION
**Pull request by eshelB**
Added operations + tests of:
Div
Gt
Gte
Rem (remainder)
Shl (shift left)
Shr (shift right)
Max
Min
Eq
Ne
And
Xor
Or

On another branch: Not - since it's unary so it required a new abi

The only others that are left that I've seen are:

   * Neg (negation - minus) - Itzik said it's not relevant for now. Also we are currently working with unsigned numbers (if I understood the code correcly)...
   * rand?

### CODE REVIEW

**Review by Cashmaney — APPROVED** [review]

File: internal/api/amd64/lib.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ const (
 	mul               = C.Op_Mul
 	lt                = C.Op_Lt
 	lte               = C.Op_Lte
+	// todo why is this here?
```
Comment (liorbond) [inline internal/api/amd64/lib.go:46]: Please speak with @Cashmaney whether delete this code or add your ops

File: internal/api/types.go [hunk 2]
```diff
@@ -27,6 +27,20 @@ const (
...
+	and        = 9
+	or         = 10
+	xor        = 11
+	eq         = 12
+	ne         = 13
+	min        = 14
+	max        = 15
+	shl        = 16
+	shr        = 17
+	// todo add more ops
```
Comment (liorbond) [inline internal/api/types.go:43]: what ops should we add?

File: libtfhe-wrapper/src/api/ffi/api.rs [hunk 3]
```diff
@@ -29,6 +29,20 @@ pub enum Op {
...
+    BitAnd = 9,
+    BitOr = 10,
+    BitXor = 11,
+    Eq = 12,
+    Ne = 13,
+    Min = 14,
+    Max = 15,
+    Shl = 16,
+    Shr = 17,
+    // todo remaining ops
```
Comment (liorbond) [inline libtfhe-wrapper/src/api/ffi/api.rs:45]: what ops?

File: libtfhe-wrapper/src/api/ffi/api.rs [hunk 4]
```diff
@@ -39,6 +53,20 @@ impl From<u32> for Op {
...
+            9 => Op::BitAnd,
+            10 => Op::BitOr,
+            11 => Op::BitXor,
+            12 => Op::Eq,
+            13 => Op::Ne,
+            14 => Op::Min,
+            15 => Op::Max,
+            16 => Op::Shl,
+            17 => Op::Shr,
+            // todo add remaining ops
```
Comment (liorbond) [inline libtfhe-wrapper/src/api/ffi/api.rs:69]: what ops?

File: libtfhe-wrapper/src/math.rs [hunk 5]
```diff
@@ -87,6 +102,20 @@ fn common_op<
...
+        Op::BitAnd => num1 & num2,
+        Op::BitOr => num1 | num2,
+        Op::BitXor => num1 ^ num2,
+        Op::Eq => num1.eq(num2),
+        Op::Ne => num1.ne(num2),
+        Op::Min => num1.min(&num2),
+        Op::Max => num1.max(&num2),
+        Op::Shl => num1 << num2,
+        Op::Shr => num1 >> num2,
+        // todo add remaining ops
```
Comment (liorbond) [inline libtfhe-wrapper/src/math.rs:118]: what ops?

**Review by liorbond — CHANGES_REQUESTED** [review]

**Pull request merged**